src/streamlink/plugins/youtube.py

```diff
@@ -214,84 +209,27 @@ def _create_adaptive_streams(self, info, streams, protected):
 
         return streams, protected
 
-    def _find_channel_video(self):
-        res = http.get(self.url)
+    def _find_video_id(self, url):
+        res = http.get(url)
+
+        for link in itertags(res.text, 'link'):
+            if link.attributes.get("rel") == "canonical":
+                canon_link = link.attributes.get("href")
+                if canon_link != url:
+                    log.debug("Re-directing to canonical URL: {0}".format(canon_link))
+                    return self._find_video_id(canon_link)
 
         datam = _ytdata_re.search(res.text)
         if datam:
             data = parse_json(datam.group(1))
             # find the videoRenderer object, where there is a LVE NOW badge
-            for x in search_dict(data, 'videoRenderer'):
```

Comment:
https://www.youtube.com/user/france24
https://www.youtube.com/user/ROCKETBEANSTV
https://www.youtube.com/user/deutschewelleenglish

This channels only have `videoRenderer` not `currentVideoEndpoint`

---

only `/live` has `currentVideoEndpoint`
https://www.youtube.com/user/france24/live

Reply:
Regular videos have it too. `channelVideoPlayerRenderer` seems to be a good option for non-`/live` URLs.

Reply:
`channelVideoPlayerRenderer` will open a vod on

https://www.youtube.com/user/france24

```
'channelVideoPlayerRenderer': {
    'videoId': 'd2xCh1hRwT4',
```

Reply:
True, then `videoRenderer` as in the current version.